File: basetoolslibrary/src/main/java/com/excellence/basetoolslibrary/utils/CloseUtils.java
package com.excellence.basetoolslibrary.utils;

import android.support.annotation.NonNull;

import java.io.Closeable;
import java.io.IOException;

/**
 * <pre>
 *     author : VeiZhang
 *     blog   : https://veizhang.github.io/
 *     time   : 2017/3/10
 *     desc   : 关闭相关工具类
 * </pre>
 */

public class CloseUtils
{
	/**
	 * 关闭IO
	 *
	 * @param closeables closeable
	 */
	public static void closeIO(@NonNull Closeable... closeables)
	{
		for (Closeable closeable : closeables)
		{
			if (closeable != null)
			{
				try
				{
					closeable.close();
				}
				catch (IOException e)
				{
					e.printStackTrace();
				}
			}
		}
	}

	/**
	 * 安静关闭IO
	 *
	 * @param closeables closeable
	 */
	public static void closeIOQuietly(@NonNull Closeable... closeables)
	{
		for (Closeable closeable : closeables)
		{
			if (closeable != null)
			{
				try
				{
					closeable.close();
				}
				catch (IOException ignored)
				{

				}
			}
		}
	}
}

File: basetoolslibrary/src/main/java/com/excellence/basetoolslibrary/utils/CloseUtils.kt
package com.excellence.basetoolslibrary.utils

import java.io.Closeable
import java.net.Socket

/**
 * <pre>
 *     author : VeiZhang
 *     blog   : http://tiimor.cn
 *     time   : 2022/4/6
 *     desc   : 关闭相关工具类
 * </pre>
 */
object CloseUtils {

    /**
     * 关闭IO
     */
    @JvmStatic
    fun closeIO(vararg closeables: Closeable) {
        for (closeable in closeables) {
            try {
                closeable.close()
            } catch (e: java.lang.Exception) {
                e.printStackTrace()
            }
        }
    }

    /**
     * 安静关闭IO
     */
    @JvmStatic
    fun closeIOQuietly(vararg closeables: Closeable) {
        for (closeable in closeables) {
            try {
                closeable.close()
            } catch (ignored: java.lang.Exception) {

            }
        }
    }

    /**
     * 关闭IO
     */
    @JvmStatic
    fun closeIO(vararg closeables: Socket) {
        for (closeable in closeables) {
            try {
                closeable.close()
            } catch (e: java.lang.Exception) {
                e.printStackTrace()
            }
        }
    }

    /**
     * 安静关闭IO
     */
    @JvmStatic
    fun closeIOQuietly(vararg closeables: Socket) {
        for (closeable in closeables) {
            try {
                closeable.close()
            } catch (ignored: java.lang.Exception) {

            }
        }
    }
}